File: kala-game-service/src/main/java/org/jesperancinha/games/kalagameservice/service/GameService.java
package org.jesperancinha.games.kalagameservice.service;

import org.jesperancinha.games.kalagameservice.model.Board;
import org.jesperancinha.games.kalagameservice.model.Pit;
import org.jesperancinha.games.kalagameservice.model.Player;

public interface GameService {
    Board createNewBoard(final Player player);

    Board sowStonesFromPit(final Player player, final Pit pit, final Board board);

    Board joinPlayer(final Player player, final Board board);
}

File: kala-game-service/src/main/kotlin/org/jesperancinha/games/kalagameservice/service/GameService.kt
package org.jesperancinha.games.kalagameservice.service

import org.jesperancinha.games.kalagameservice.model.Board
import org.jesperancinha.games.kalagameservice.model.Pit
import org.jesperancinha.games.kalagameservice.model.Player

interface GameService {
    fun createNewBoard(player: Player): Board?
    fun sowStonesFromPit(player: Player, pit: Pit, board: Board?): Board?
    fun joinPlayer(player: Player, board: Board?): Board?
}